spark/src/main/scala/org/apache/spark/sql/delta/actions/TableFeatureSupport.scala

```diff
@@ -281,12 +281,34 @@ trait TableFeatureSupport { this: Protocol =>
     (this.readerAndWriterFeatureNames -- to.readerAndWriterFeatureNames).size == 1
   }
 
+  /**
+   * This is a special protocol version downgrade path that can be invoked by setting legacy
+   * protocol versions on a table features protocol. This is the only downgrade path that may
+   * occur outside the DROP FEATURE command. The main requirement is the table features protocol
+   * contains only legacy features.
+   *
+   * During downgrade, we first add the relevant implicit legacy features to the protocol
+   * and then call downgradeProtocolVersionsIfNeeded. At this stage, the validation stage,
+   * we only have available the the start and end states of the Protocol, i.e. the intermediate
+   * state with all the required implicit features is not available. It is
+   * downgradeProtocolToLegacyVersionsIfNeeded which validates whether the intermediate state has
+   * all required legacy features are in the protocol before downgrading the protocol versions.
+   */
+  def canDowngradeToLegacy(to: Protocol): Boolean = {
+    require(supportsWriterFeatures)
+    require(!to.supportsWriterFeatures)
+
+    readerAndWriterFeatures.forall(_.isLegacyFeature) &&
+      readerAndWriterFeatureNames.subsetOf(to.implicitlySupportedFeatures.map(_.name))
+  }
+
   /**
    * True if this protocol can be upgraded or downgraded to the 'to' protocol.
    */
   def canTransitionTo(to: Protocol, op: Operation): Boolean = {
     op match {
       case drop: DeltaOperations.DropTableFeature => canDowngradeTo(to, drop.featureName)
+      case _ if supportsWriterFeatures && !to.supportsWriterFeatures => canDowngradeToLegacy(to)
```

Comment:
```suggestion
      case _ if supportsWriterFeatures && !to.supportsWriterFeatures =>
        // This is called a "downgrade" but it can actually only be a "side-grade"
        // (equivalent features, different representation) or an upgrade (more
        // features, lower legacy protocol version).
        canDowngradeToLegacy(to)
```